File: LeetCode-Java/src/Link/leetcode/easy/ListNode.java
package Link.leetcode.easy;

public class ListNode {
    public int val;
    public ListNode next;
    public ListNode(int x) { val = x; }

    public static ListNode createWithNums(int[] nums) {
        ListNode head = null;
        ListNode index = head;
        for (int i = 0; i < nums.length; i++) {
            ListNode current = new ListNode(nums[i]);
            if (index != null) {
                index.next = current;
            } else {
                head = current;
            }
            index = current;
        }
        return head;
    }
}

File: LeetCodeSwift/LeetCodeSwift/Base Data Struct/Link/ListNode.swift
//
//  ListNode.swift
//  LeetCode-Swift
//
//  Created by Liang on 2020/5/1.
//  Copyright © 2020 imieutan. All rights reserved.
//

public class Link {
    public let head: ListNode

    public init(_ nums: [Int]) {
        let dummyNode = ListNode(-1)
        var indexNode = dummyNode
        nums.forEach { (num) in
            let currentNode = ListNode(num)
            indexNode.next = currentNode
            indexNode = currentNode
        }
        head = dummyNode.next!
    }
}

public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init(_ val: Int) {
        self.val = val
        self.next = nil
    }

    public init(_ val: Int, next: ListNode?) {
        self.val = val
        self.next = next
    }
}
